packages/flutter_test/lib/src/binding.dart

```diff
@@ -488,19 +489,26 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
   /// events from the device).
   Offset localToGlobal(Offset point) => point;
 
+  /// The source of the current pointer event.
+  ///
+  /// The [pointerEventSource] is set as the `source` parameter of
+  /// [handlePointerEvent] and can be used in the immediate enclosing
+  /// [dispatchEvent].
+  @protected
+  TestBindingEventSource pointerEventSource = TestBindingEventSource.device;
+
   @override
-  void dispatchEvent(
```

Comment:
Presumably, we would want to keep this override to ensure that only `TestBindingEventSource.test` reach this method?

Reply:
It's something I've thought for a long time, alnd I don't think this design is either sufficient or necessary to achieve what it seems to.
Redefining `TestWidgetsFlutterBinding.dispatchEvent` provides a way to diverge the behavior of `dispatchEvent` based on event source. The new behavior is only implemented by `LiveTestWidgetsFlutterBinding`. How it behaves based on the source is shown as follows:
```
// TestWidgetsFlutterBinding binding;
binding.dispatchEvent(event, result)  // AssertionError
binding.dispatchEvent(event, result, source: TestBindingEventSource.test)  // Test behavior
(binding as GestureBinding).dispatchEvent(event, result)  // Original behavior
```
But this is why the assertion is unnecessary: If some class calls `dispatchEvent` without adding `source`, the class must be unaware of `TestWidgetsFlutterBinding`, which means it will always call the method as `GestureBinding.dispatchEvent`. This assertion protects no one.

And why shouldn't any class aware of `TestWidgetsFlutterBinding` call the device version? The `LiveTestWidgetsFlutterBinding.dispatchEvent` even defines its own way of handling device-sourced events. The only `dispatchEvent` that does not accept device-sourced events is `TestWidgetsFlutterBinding`'s, which itself does not even redefine test-sourced events. If we really want to diverge the sources, we better do it in `LiveTestWidgetsFlutterBinding` (which does not need it either, since it _allows_ device-sourced events).

What's worse, it forces a test in `packages/flutter/test/gestures/gesture_binding_resample_event_on_widget_test.dart`  to shadow the argument.

Since the assertion is not needed, and the divergence is already implemented by `handlePointerEvent`, I can really think of no reason to keep this assertion.